<a href="https://colab.research.google.com/github/niltontac/EspAnalise-EngDados/blob/master/Novel_Coronavirus_analysis_and_predictions_2019_2020%20-%20In%20Progress.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Sobre este conjunto de dados

#####Este counjunto de dados possui informações diárias sobre os números de casos afetados, mortes e recuperações do novo Corona vírus de 2019. Observe que esses são dados de séries temporais e, portando, os números de casos em um determinado dia são números acumulados.


#About this Dataset

#####This dataset has daily level information on the numbers affected cases, deaths and recoveries from the new coronavirus. Note that these are data from time series and the numbers of cases on a given day are cumulative numbers.

---

#####Fonte | Source (Dataset): https://www.kaggle.com/sudalairajkumar/novel-corona-virus-2019-dataset

---

#####Analyst: Nilton Thiago de Andrade Coura


In [0]:
# Importing Libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Loading dataset

dataset = pd.read_csv('https://raw.githubusercontent.com/niltontac/EspAnalise-EngDados/master/data/Novel_Corona_Virus_2019_Dataset/covid_19_data.csv', parse_dates=['ObservationDate', 'Last Update'])

In [2]:
# Dataset Dimension (rows x columns)

dataset.shape

(4513, 8)

In [3]:
# Dataset Information

dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4513 entries, 0 to 4512
Data columns (total 8 columns):
SNo                4513 non-null int64
ObservationDate    4513 non-null datetime64[ns]
Province/State     2912 non-null object
Country/Region     4513 non-null object
Last Update        4513 non-null datetime64[ns]
Confirmed          4513 non-null float64
Deaths             4513 non-null float64
Recovered          4513 non-null float64
dtypes: datetime64[ns](2), float64(3), int64(1), object(2)
memory usage: 282.2+ KB


In [4]:
# Dataset describe

dataset.describe().T

count         mean          std  ...     50%     75%      max
SNo        4513.0  2257.000000  1302.935212  ...  2257.0  3385.0   4513.0
Confirmed  4513.0   577.460891  4989.693765  ...     8.0    93.0  67743.0
Deaths     4513.0    17.382451   190.066018  ...     0.0     1.0   3008.0
Recovered  4513.0   190.690228  2038.424657  ...     1.0    15.0  46488.0

[4 rows x 8 columns]

In [5]:
# Dataset (columns data types) 

dataset.dtypes

SNo                         int64
ObservationDate    datetime64[ns]
Province/State             object
Country/Region             object
Last Update        datetime64[ns]
Confirmed                 float64
Deaths                    float64
Recovered                 float64
dtype: object

In [6]:
# Checking the last 5 cases to see when the dataset was updated

dataset.tail().style.background_gradient(cmap='PRGn')

In [7]:
# The countries list

", ".join(dataset['Country/Region'].unique().tolist())

'Mainland China, Hong Kong, Macau, Taiwan, US, Japan, Thailand, South Korea, Singapore, Philippines, Malaysia, Vietnam, Australia, Mexico, Brazil, Colombia, France, Nepal, Canada, Cambodia, Sri Lanka, Ivory Coast, Germany, Finland, United Arab Emirates, India, Italy, UK, Russia, Sweden, Spain, Belgium, Others, Egypt, Iran, Israel, Lebanon, Iraq, Oman, Afghanistan, Bahrain, Kuwait, Austria, Algeria, Croatia, Switzerland, Pakistan, Georgia, Greece, North Macedonia, Norway, Romania, Denmark, Estonia, Netherlands, San Marino,  Azerbaijan, Belarus, Iceland, Lithuania, New Zealand, Nigeria, North Ireland, Ireland, Luxembourg, Monaco, Qatar, Ecuador, Azerbaijan, Czech Republic, Armenia, Dominican Republic, Indonesia, Portugal, Andorra, Latvia, Morocco, Saudi Arabia, Senegal, Argentina, Chile, Jordan, Ukraine, Saint Barthelemy, Hungary, Faroe Islands, Gibraltar, Liechtenstein, Poland, Tunisia, Palestine, Bosnia and Herzegovina, Slovenia, South Africa, Bhutan, Cameroon, Costa Rica, Peru, Serbia

In [0]:
# Rename some countries to make sure all adhere to (alpha-3) ISO-Standards and for the ploty
# map visualization to work

dataset['Country/Region'] = dataset['Country/Region'].replace('US', 'USA')
dataset['Country/Region'] = dataset['Country/Region'].replace('South Korea', 'KOR')
dataset['Country/Region'] = dataset['Country/Region'].replace('Russia', 'RUS')
dataset['Country/Region'] = dataset['Country/Region'].replace('UK', 'GBR')


# Rename the columns

dataset = dataset.rename(columns={'Country/Region': 'Country', 'ObservationDate': 'Date'})

In [9]:
# Removing unessecary columns

dataset = dataset.drop(['SNo', 'Last Update'], axis=1)

# Checking

dataset.tail().style.background_gradient(cmap='PRGn')

In [10]:
# Checking for null or missing values

pd.DataFrame(dataset.isnull().sum()).T

Date  Province/State  Country  Confirmed  Deaths  Recovered
0     0            1601        0          0       0          0

The dataset has 1601 missings values in "Province/State" column.
Let's replace them with 'unknow'

In [0]:
# replacing data missings

dataset = dataset.fillna('unknow')

In [12]:
# Checking for null or missing values again

pd.DataFrame(dataset.isnull().sum()).T

Date  Province/State  Country  Confirmed  Deaths  Recovered
0     0               0        0          0       0          0